pkg/apis/core/types.go

```diff
@@ -3674,6 +3693,16 @@ type ServiceSpec struct {
 	// +optional
 	ExternalTrafficPolicy ServiceExternalTrafficPolicyType
 
+	// InternalTrafficPolicy specifies if the cluster internal traffic
+	// should be routed to cluster-wide endpoints, node-local endpoints only
+	// or both with node-local ones preferred if there is any. "Cluster"
+	// routes internal traffic to a Service to all cluster-wide endpoints.
+	// "PreferLocal" will route internal traffic to node-local endpoints if
+	// one exists, otherwise it will fallback to the same behavior as
+	// "Cluster". "Local" routes traffic to node-local endpoints only,
+	// traffic is dropped if no node-local endpoints are ready.
+	InternalTrafficPolicy ServiceInternalTrafficPolicyType
```

Comment:
We need to also update the v1 types: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L4032-L4253

And run `make update` to update generated code.